--- a/ld/emultempl/pe.em	2021-01-09 11:47:34.000000000 +0100
+++ b/ld/emultempl/pe.em	2021-02-28 09:40:12.547395700 +0100
@@ -104,8 +104,12 @@ fragment <<EOF
 #define DEFAULT_PSEUDO_RELOC_VERSION 1
 #endif
 
+#if defined(__CYGWIN__) || defined (__MINGW32__)
+#define DEFAULT_DLL_CHARACTERISTICS	(0)
+#else
 #define DEFAULT_DLL_CHARACTERISTICS	(IMAGE_DLL_CHARACTERISTICS_DYNAMIC_BASE \
 					 | IMAGE_DLL_CHARACTERISTICS_NX_COMPAT)
+#endif
 
 #if defined(TARGET_IS_i386pe) || ! defined(DLL_SUPPORT)
 #define	PE_DEF_SUBSYSTEM		3
--- a/ld/emultempl/pep.em	2021-01-09 11:47:34.000000000 +0100
+++ b/ld/emultempl/pep.em	2021-02-28 09:40:52.328116000 +0100
@@ -99,9 +99,13 @@ fragment <<EOF
 #define DLL_SUPPORT
 #endif
 
+#if defined(__CYGWIN__) || defined (__MINGW32__)
+#define DEFAULT_DLL_CHARACTERISTICS	(0)
+#else
 #define DEFAULT_DLL_CHARACTERISTICS	(IMAGE_DLL_CHARACTERISTICS_DYNAMIC_BASE \
 					 | IMAGE_DLL_CHARACTERISTICS_HIGH_ENTROPY_VA \
 					 | IMAGE_DLL_CHARACTERISTICS_NX_COMPAT)
+#endif
 
 #if defined(TARGET_IS_i386pep) || ! defined(DLL_SUPPORT)
 #define	PE_DEF_SUBSYSTEM		3
